src/architecture/principles.md

```diff
@@ -0,0 +1,287 @@
+- That JS piece of code in the first bullet can be turned into an IPFS implementation if it
+  fetches from a :cite[trustless-gateway] and verifies what it gets.
+
+## Self-Certifying Addressability
+
+:dfn[Authority] is control over a given domain of competence and :dfn[naming authority] is
+control over what resources are called.
+
+:dfn[Addressability] is the property of a naming system such that its names are sufficient
+for an agent to interact with the resources being named.
+
+:dfn[Verifiability] is the property of a naming system such that an agent can certify that
+the mapping between a name it uses and a resource it is interacting with is correct without
+recourse to an authority other than itself and the resource.
+
+:dfn[Self-certifying addressability] is the property of a naming system such that it is both
+addressable and verifiable: any name is sufficient to interact with a resource, and its mapping
+to that resource can be certified without recourse to additional authority. Self-certifying
+addressability is a key component of a
+[self-certifying web](https://jaygraber.medium.com/web3-is-self-certifying-9dad77fd8d81)
+and it supports capture-resistance which can help mitigate against centralization.
+
+CIDs support :ref[self-certifying addressability]. With CIDs, the authority to name a resource
+resides only with that resource and derives directly from that resource's most intrinsic
+property: its content. This frees interactions with CID-named resources from the power
+relation implicit in a client-server architecture. CIDs are the trust model of IPFS.
+
+An implementation may retrieve a CID without verifying that the resource matches it, but that
+loses the resource's naming authority. Such an implementation would be comparable to an HTTP
+client looking DNS records up from a random person's resolver: it cannot guarantee that the
+addressing is authoritative. Implementers may make informed decisions as to where in their
+systems they support verification, but they should ensure that CIDs are verified whenever
+they also have access to the resource that the CID maps to.
+
+## Appendix: Historical Notes
+
+We tend not to think about addressability because it is so foundational that we
+struggle to apprehend a system without it, but that is precisely why it is important
+that we get it right. You can find extensive historical evidence that TimBL and others saw
+URLs as arguably the most fundamental invention of the Web, and the early groups that
+worked on Web architecture discussed and debated the properties of URLs at length. The
+problems of centralization we face today trace their lineage back to those decisions.
+
+The hierarchical nature of the HTTP addresses was intentional, as TimBL wrote clearly in
+[Web Architecture from 50,000 feet](https://www.w3.org/DesignIssues/Architecture.html):
+> The HTTP space consists of two parts, one hierarchically delegated, for which the
```

Comment:
aha, here's `space`. maybe reference this explicitly above in the `IPFS space` comment, in addition to the fixes requested there.

Reply:
I think it's extraneous. This is provided for historical context, but it's not like "HTTP space" is used that much. We don't need to copy things that Tim said forever ago :)